pinecone_text/dense/openai_encoder.py

```diff
@@ -14,16 +15,18 @@ class OpenAIEncoder(BaseDenseEncoder):
     OpenAI's text embedding wrapper. See https://platform.openai.com/docs/guides/embeddings
 
     Note: You should provide an API key and organization in the environment variables OPENAI_API_KEY and OPENAI_ORG.
+          Or you can pass them as arguments to the constructor as `api_key` and `organization`.
     """
 
-    def __init__(self, model_name: str = "text-embedding-ada-002"):
+    def __init__(self, model_name: str = "text-embedding-ada-002", **kwargs: Any):
```

Comment:
@igiloh-pinecone I'm not very sure about the kwargs here. The reason I chose kwargs is that OpenAI have some object called [NOT_GIVEN](https://github.com/openai/openai-python/blob/e661da69b4a11d48edfe21d2b12f53c201593596/src/openai/_client.py#L68C58-L68C67) which not allow us to get timeout as `Optoinal[int] = None` since here None might mean NOT_GIVEN or None which is a different thing w.r.t. openai

Reply:
Let's take a compromise - let's expose the 3 most important init arguments explicitly - `apy_key`, `organization`, and `base_url`. The rest will be `**kwargs`.  

(I'm referring to `pinecone_text`. In `canopy` we should probably do pure `**kwargs` like you did here)